web/src/settings_users.js

```diff
@@ -628,45 +643,42 @@ export function show_edit_user_info_modal(user_id, from_user_info_popover) {
+        owner_is_only_user_in_organization: people.get_active_human_count() === 1,
     });
+    $container.append(html_body);
+
+    // Set role dropdown and fields user pills
+    $("#user-role-select").val(person.role);
+    if (!page_params.is_owner) {
+        $("#user-role-select")
+            .find(`option[value="${CSS.escape(settings_config.user_role_values.owner.code)}"]`)
+            .hide();
+    }
 
-    let fields_user_pills;
+    const custom_profile_field_form = "#edit-user-form .custom-profile-field-form";
+    $(custom_profile_field_form).empty();
+    settings_account.append_custom_profile_fields(custom_profile_field_form, user_id);
+    settings_account.initialize_custom_date_type_fields(custom_profile_field_form);
+    const fields_user_pills = settings_account.initialize_custom_user_type_fields(
+        custom_profile_field_form,
+        user_id,
+        true,
+        false,
+    );
 
-    function set_role_dropdown_and_fields_user_pills() {
-        $("#user-role-select").val(person.role);
-        if (!page_params.is_owner) {
-            $("#user-role-select")
-                .find(`option[value="${CSS.escape(settings_config.user_role_values.owner.code)}"]`)
-                .hide();
+    // Handle deactivation
+    $("#edit-user-form").on("click", ".deactivate_user_button", (e) => {
+        e.preventDefault();
+        e.stopPropagation();
+        const user_id = $("#edit-user-form").data("user-id");
+        function handle_confirm() {
+            const url = "/json/users/" + encodeURIComponent(user_id);
+            dialog_widget.submit_api_request(channel.del, url);
         }
+        confirm_deactivation(user_id, handle_confirm, true);
+    });
 
-        const element = "#edit-user-form .custom-profile-field-form";
-        $(element).empty();
-        settings_account.append_custom_profile_fields(element, user_id);
-        settings_account.initialize_custom_date_type_fields(element);
-        fields_user_pills = settings_account.initialize_custom_user_type_fields(
-            element,
-            user_id,
-            true,
-            false,
-        );
-        settings_account.initialize_custom_pronouns_type_fields(element);
-
-        $("#edit-user-form").on("click", ".deactivate_user_button", (e) => {
-            e.preventDefault();
-            e.stopPropagation();
-            const user_id = $("#edit-user-form").data("user-id");
-            function handle_confirm() {
-                const url = "/json/users/" + encodeURIComponent(user_id);
-                dialog_widget.submit_api_request(channel.del, url);
-            }
-            const open_deactivate_modal_callback = () =>
-                confirm_deactivation(user_id, handle_confirm, true);
-            dialog_widget.close_modal(open_deactivate_modal_callback);
-        });
-    }
-
-    function submit_user_details() {
+    $("#user-profile-modal").on("click", ".dialog_submit_button", () => {
```

Comment:
Yeah. Seeing this pattern all over again makes me even more suspicious. I wonder if there's not a way to push harder here and even better align the bot and user modals.